File: Sources/CountriesFlagsObjC/source/Extensions/UIBezierPath+Rotate.m
//
//  UIBezierPath+Rotate.m
//  CountriesFlags
//

#import "Extensions/UIBezierPath+Rotate.h"

@implementation UIBezierPath (Rotate)

- (void)rotateToAngle:(CGFloat)angle center:(CGPoint)center {
    CGFloat radians = angle / 180.0f * M_PI;
    CGAffineTransform transform = CGAffineTransformIdentity;
    transform = CGAffineTransformTranslate(transform, center.x, center.y);
    transform = CGAffineTransformRotate(transform, radians);
    transform = CGAffineTransformTranslate(transform, -center.x, -center.y);
    [self  applyTransform:transform];
}

@end

File: Sources/CountriesFlags/Extensions/UIBezierPath+Rotate.swift
//
//  UIBezierPath+Rotate.swift
//  CountriesFlags
//

import UIKit

extension UIBezierPath {
    /**
     Rotates UIBezierPath around the specified center.
     
     - Parameters:
        - angle: Angle of rotation in degrees.
        - center: Center of rotation.
     */
    func rotate(to angle: CGFloat, around center: CGPoint) {
        let radians = angle / 180.0 * .pi
        var transform: CGAffineTransform = .identity
        transform = transform.translatedBy(x: center.x, y: center.y)
        transform = transform.rotated(by: radians)
        transform = transform.translatedBy(x: -center.x, y: -center.y)
        apply(transform)
    }
}
